cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/DefaultFlowConfigurer.java

```diff
@@ -78,23 +79,29 @@ public void addFlowlet(String name, Flowlet flowlet) {
   }
 
   private String getFlowletName(Flowlet flowlet) {
-    FlowletDefinition flowletDef = new FlowletDefinition(null, flowlet, 1);
+    DefaultFlowletConfigurer flowletConfigurer = new DefaultFlowletConfigurer(flowlet);
+    flowlet.configure(flowletConfigurer);
+    FlowletSpecification flowletSpecification = flowletConfigurer.createSpecification();
+    FlowletDefinition flowletDef = new FlowletDefinition(null, flowlet, flowletSpecification, 1);
```

Comment:
don't think you need this line, can just get it from the flowletSpecification. FlowletDefinition does some field injection, but it doesn't seem like that's needed.

Reply:
updated it